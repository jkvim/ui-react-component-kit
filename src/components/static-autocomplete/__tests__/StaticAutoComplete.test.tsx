import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { StaticAutoComplete } from '../';

describe('StaticAutoComplete', () => {

  const option = { display: 'Test Display', value: 'Test Value'};

  describe('props', () => {

    const mockCallback = jest.fn();

    const wrapper = mount(
      <StaticAutoComplete
        id="id"
        label="label"
        placeholder="placeholder"
        suggestions={[option]}
        selectedItem={{value: 'Test Value', item: option}}
        valuePath="value"
        hint="hint"
        disabled={true}
        errorMessage="message"
        className="class"
        onBlur={mockCallback}
        onChange={mockCallback}
      />);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `id` in props', () => {
      expect(wrapper.props().id).toBe('id');
    });

    it('should pass `label` in props', () => {
      expect(wrapper.props().label).toBe('label');
    });

    it('should pass `placeholder` in props', () => {
      expect(wrapper.props().placeholder).toBe('placeholder');
    });

    it('should pass `suggestions` in props', () => {
      expect(wrapper.props().suggestions).toEqual([option]);
    });

    it('should pass `selectedItem` in props', () => {
      expect(wrapper.props().selectedItem).toEqual({value: 'Test Value', item: option});
    });

    it('should pass `valuePath` in props', () => {
      expect(wrapper.props().valuePath).toBe('value');
    });

    it('should pass `hint` in props', () => {
      expect(wrapper.props().hint).toBe('hint');
    });

    it('should pass `disabled` in props', () => {
      expect(wrapper.props().disabled).toBe(true);
    });

    it('should pass `errorMessage` in props', () => {
      expect(wrapper.props().errorMessage).toBe('message');
    });

    it('should pass `className` in props', () => {
      expect(wrapper.props().className).toBe('class');
    });

    it('should pass `onBlur` in props', () => {
      expect(wrapper.props().onBlur).toBe(mockCallback);
    });

    it('should pass `onChange` in props', () => {
      expect(wrapper.props().onChange).toBe(mockCallback);
    });
  });

  it('snapshot basic', () => {
    const wrapper = mount(
      <StaticAutoComplete id="id" label="label" suggestions={[option]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot error', () => {
    const wrapper = mount(
      <StaticAutoComplete id="id" label="label" suggestions={[option]} errorMessage="message" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot disabled', () => {
    const wrapper = mount(
      <StaticAutoComplete id="id" label="label" suggestions={[option]} disabled={true} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});