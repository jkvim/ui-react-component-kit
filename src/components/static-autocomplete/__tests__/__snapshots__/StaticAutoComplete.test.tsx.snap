// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticAutoComplete should match snapshot 1`] = `
<StaticAutoComplete
  id="static-id"
  label="Label"
  suggestions={
    Array [
      "one",
      "two",
      "three",
    ]
  }
>
  <AutoComplete
    dataProvider={
      StaticDataProvider {
        "suggestions": Array [
          "one",
          "two",
          "three",
        ],
        "valuePath": undefined,
      }
    }
    id="static-id"
    label="Label"
    onBlur={[Function]}
    onChange={[Function]}
  >
    <styled.div
      onBlur={[Function]}
      role="application"
    >
      <div
        className="sc-iwsKbI jbdQuT"
        onBlur={[Function]}
        role="application"
      >
        <InputText
          aria-activedescendant=""
          aria-autocomplete="list"
          aria-describedby="static-id-autocomplete-error"
          aria-expanded={false}
          aria-invalid={false}
          aria-owns="static-id-suggestions"
          id="static-id-autocomplete"
          label="Label"
          onChange={[Function]}
          onKeyDown={[Function]}
          role="combobox"
          setInputRef={[Function]}
          showError={false}
          value=""
        >
          <styled.div>
            <div
              className="sc-ifAKCX cnyXj"
            >
              <styled.label
                hasError={false}
                htmlFor="static-id-autocomplete"
                theme={
                  Object {
                    "border": Object {
                      "radius": Object {
                        "edge": "2px",
                        "round": "100px",
                      },
                      "size": "2px",
                    },
                    "breakpoint": Object {
                      "lg": "992px",
                      "md": "768px",
                      "sm": "576px",
                      "xl": "1200px",
                      "xs": "0px",
                    },
                    "color": Object {
                      "background": "#ffffff",
                      "border": "#2fa4e7",
                      "checked": "#2fa4e7",
                      "danger": "#c71c22",
                      "default": "#2fa4e7",
                      "disabled": "#acacac",
                      "error": "#c71c22",
                      "focus": "#54b4eb",
                      "font": "#333333",
                      "hover": "#178acc",
                      "info": "#04519b",
                      "primary": "#2fa4e7",
                      "secondary": "#e9ecef",
                      "success": "#73a839",
                      "tertiary": "#343a40",
                      "warning": "#dd5600",
                    },
                    "font": Object {
                      "size": Object {
                        "default": "1.8rem",
                      },
                      "weight": Object {
                        "default": 100,
                      },
                    },
                  }
                }
              >
                <label
                  className="sc-bdVaJa mlfRZ"
                  htmlFor="static-id-autocomplete"
                >
                  Label
                </label>
              </styled.label>
              <styled.input
                aria-describedby="static-id-autocomplete-error"
                aria-invalid={false}
                hasError={false}
                id="static-id-autocomplete"
                innerRef={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder=""
                theme={
                  Object {
                    "border": Object {
                      "radius": Object {
                        "edge": "2px",
                        "round": "100px",
                      },
                      "size": "2px",
                    },
                    "breakpoint": Object {
                      "lg": "992px",
                      "md": "768px",
                      "sm": "576px",
                      "xl": "1200px",
                      "xs": "0px",
                    },
                    "color": Object {
                      "background": "#ffffff",
                      "border": "#2fa4e7",
                      "checked": "#2fa4e7",
                      "danger": "#c71c22",
                      "default": "#2fa4e7",
                      "disabled": "#acacac",
                      "error": "#c71c22",
                      "focus": "#54b4eb",
                      "font": "#333333",
                      "hover": "#178acc",
                      "info": "#04519b",
                      "primary": "#2fa4e7",
                      "secondary": "#e9ecef",
                      "success": "#73a839",
                      "tertiary": "#343a40",
                      "warning": "#dd5600",
                    },
                    "font": Object {
                      "size": Object {
                        "default": "1.8rem",
                      },
                      "weight": Object {
                        "default": 100,
                      },
                    },
                  }
                }
                type="text"
                value=""
              >
                <input
                  aria-describedby="static-id-autocomplete-error"
                  aria-invalid={false}
                  className="sc-dnqmqq ceLtZr"
                  id="static-id-autocomplete"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder=""
                  type="text"
                  value=""
                />
              </styled.input>
            </div>
          </styled.div>
        </InputText>
        <styled.div
          aria-live="polite"
        >
          <div
            aria-live="polite"
            className="sc-bZQynM jbtBHc"
          />
        </styled.div>
        <styled.div
          hasError={false}
          id="static-id-autocomplete-error"
          theme={
            Object {
              "border": Object {
                "radius": Object {
                  "edge": "2px",
                  "round": "100px",
                },
                "size": "2px",
              },
              "breakpoint": Object {
                "lg": "992px",
                "md": "768px",
                "sm": "576px",
                "xl": "1200px",
                "xs": "0px",
              },
              "color": Object {
                "background": "#ffffff",
                "border": "#2fa4e7",
                "checked": "#2fa4e7",
                "danger": "#c71c22",
                "default": "#2fa4e7",
                "disabled": "#acacac",
                "error": "#c71c22",
                "focus": "#54b4eb",
                "font": "#333333",
                "hover": "#178acc",
                "info": "#04519b",
                "primary": "#2fa4e7",
                "secondary": "#e9ecef",
                "success": "#73a839",
                "tertiary": "#343a40",
                "warning": "#dd5600",
              },
              "font": Object {
                "size": Object {
                  "default": "1.8rem",
                },
                "weight": Object {
                  "default": 100,
                },
              },
            }
          }
        >
          <div
            className="sc-htoDjs keLCLG"
            id="static-id-autocomplete-error"
          />
        </styled.div>
      </div>
    </styled.div>
  </AutoComplete>
</StaticAutoComplete>
`;
